src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java

```diff
@@ -122,36 +133,176 @@ private boolean isVendorOwnershipCodeApplicableForTaxIdRoute(final VendorDetail
+                            newVendorDetail.getVendorNumber(), employeeId);
+                } else {
+                    LOG.debug("addSearchAliasAndSaveVendorDetailIfNeeded, vendor {}, updating alias named {} to active",
+                            newVendorDetail.getVendorNumber(), employeeId);
+                    alias.setActive(true);
+                    saveDocumentInNewGlobalVariables(document);
+                }
+            }
+        }
+    }
+    
+    private String findEmployeeId(WorkdayKfsVendorLookupRoot root) {
+        String employeeId = StringUtils.EMPTY;
+        if (root == null || root.getResults().get(0) == null) {
+            LOG.error("findEmployeeId, for document {}, WorkdayKfsVendorLookupRoot was null, or result list was empty, this should not happen",
+                    getDocumentNumber());
+        } else {
+            WorkdayKfsVendorLookupResult result = root.getResults().get(0);
+            employeeId = result.getEmployeeID();
+            if (StringUtils.isBlank(employeeId)) {
+                LOG.error("findEmployeeId, for document {}, there was a Workday response, but no employee id was provided, this should not happen",
+                        getDocumentNumber());
+            }
+        }
+        return employeeId;
+    }
+    
+    private VendorAlias findEmployeeIdAlias(VendorDetail vendorDetail, String employeeId) {
+        for (VendorAlias alias : vendorDetail.getVendorAliases()) {
+            if (StringUtils.equalsIgnoreCase(employeeId, alias.getVendorAliasName())) {
+                return alias;
+            }
+        }
+        return null;
+    }
+    
+    private void addEmployeeIdAliasToNewVendorDetail(VendorDetail newVendorDetail, String employeeId) {
+        VendorAlias alias = new VendorAlias();
+        alias.setActive(true);
+        alias.setVendorAliasName(employeeId);
+        alias.setNewCollectionRecord(true);
+        newVendorDetail.getVendorAliases().add(alias);
+    }
+
+    private void addBlankAliasToOldVendorDetail(VendorDetail oldVendorDetail) {
+        /*
+         * A blank alias needs to be added to the old vendor detail to keep the alias List sizes in synch between 
+         * the old and new vendor detail detail on the maintenance document.
+         */
+        VendorAlias blankAlias = new VendorAlias();
+        blankAlias.setActive(false);
+        blankAlias.setNewCollectionRecord(true);
+        oldVendorDetail.getVendorAliases().add(blankAlias);
+    }
+    
+    private void saveDocumentInNewGlobalVariables(MaintenanceDocument document) {
+        try {
+            GlobalVariables.doInNewGlobalVariables(new UserSession(KFSConstants.SYSTEM_USER),
+                () -> {
+                    try {
+                        getDocumentService().saveDocument(document, DocumentSystemSaveEvent.class);
+                        return document;
+                    } catch (Exception e) {
```

Comment:
Is this `catch` necessary, given that you're already catching the exception thrown by `doInNewGlobalVariables()`?

Reply:
No, I guess it is not, I shall remove it